content/en/blog/2022/knative.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Distributed tracing in Knative
+date: 2022-04-12
+author: Pavol Loffay
+---
+
+In this article, you will learn how distributed tracing works in Knative and we will explore how the OpenTelemetry project can make tracing support in this environment easier. We will explore Knative under the hood to understand what distributed tracing capabilities it provides out-of-the-box and which parts of the system need additional instrumentation.
+
+In this article we will not cover Knative fundamentals, please refer to the [Knative documentation](https://knative.dev/docs/) to get familiar with the project.
+
+## About Knative
+
+Knative is a serverless platform built on top of Kubernetes as a set of `CustomResourceDefinitions` (CRDs). The project is split into two logical parts: serving and eventing. Serving facilitates the creation, deployment and scaling of workload/services, and eventing facilitates event-driven communication between workloads to enable loosely coupled architectures.
+
+### Knative data flow
+
+Before we deep dive into tracing let's take a look at a data flow example. It will help us to understand Knative architecture and which parts of the system need to be instrumented in order to understand the timing characteristics of the request or transaction. On the diagram below there are two user workloads (first and second) and an incoming request marked as (1. HTTP) that goes to use workload first and then to the workload second as a cloud event message.
+
+![Knative data flow](/img/blog-knative/knative-data-flow.jpg)
```

Comment:
it helps people who can not see the picture to use the alt text to provide a description:

```markdown
![Http talks to Knative layer, which talks to the "user workload first" via http ...](/img/blog-knative/knative-data-flow.jpg)
```